app/controllers/concerns/common_commentable.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module CommonCommentable
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :set_commentable
+    before_action :set_comment, only: %i[edit update destroy]
+    before_action -> { check_authorization(@comment) }, only: %i[edit update destroy]
+  end
+
+  def edit
+    render 'comments/edit'
+  end
+
+  def create
+    @comment = @commentable.comments.new(comment_params)
+    @comment.user_id = current_user.id
```

Comment:
id ではなくオブジェクトを直接代入できます。

```suggestion
    @comment.user = current_user
```

Reply:
idにこだわってしまっていました。たしかにそうですね。修正しました。